src/main/java/seedu/address/model/scheduler/time/TimePeriod.java

```diff
@@ -44,6 +44,20 @@ public boolean isStraightAfter(TimePeriod otherTimePeriod) {
                 && isSameDay(otherTimePeriod);
     }
 
+    /**
+     * Checks if another TimePeriod has clashes.
+     */
+    public boolean hasClash(TimePeriod other) {
+        if (other.getSchoolDay().equals(this.schoolDay)) {
+            return (startTime.isBefore(other.getEndTime()) && endTime.isAfter(other.getEndTime()))
+                || (startTime.isBefore(other.getStartTime()) && endTime.isAfter(other.getStartTime()))
+                || (startTime.isAfter(other.getStartTime())) && endTime.isBefore(other.getEndTime())
+                || (startTime.isEqual(other.getStartTime()))
+                || (endTime.isEqual(other.getEndTime()));
+        }
+        return false;
+    }
```

Comment:
I think the shorter path would be the `return false` part. This has good use of happy path, but it could be better :)